src/modules/kazoo/kazoo.c

```diff
@@ -312,7 +324,30 @@ static int mod_init(void) {
 		}
     }
 
-
+    sr_kemi_eng_t *keng = NULL;
+	  if(kazoo_event_callback.s!=NULL && kazoo_event_callback.len>0) {
+		  keng = sr_kemi_eng_get();
+		  if(keng==NULL) {
+			  LM_ERR("failed to find kemi engine\n");
+			  return -1;
+		  }
+		  kazoo_route_no=-1;
+	  } 
+    else {
+		  route_no=route_lookup(&event_rt, "kazoo:consumer-event");
+		  if (route_no==-1)
+		  {
+			  LM_ERR("failed to find event_route[kazoo:consumer-event]\n");
+			  return -1;
+		  }
+		  if (event_rt.rlist[route_no]==0)
+		  {
+			  LM_WARN("event_route[kazoo:consumer-event] is empty\n");
+		  }
+		  kazoo_route_no=route_no;
+    }
+    
+    
```

Comment:
please use a different logic here. `kazoo:consumer-event` is not mandatory. you could set a variable like `kz_use_kemi` that is set if `kazoo_event_callback` is set and `kemi` is valid. then use that variable when needed

Reply:
Now removed route_lookup for kazoo:consumer-event as it may or may not exist.